### ISSUE TITLE
test failure in release v1.1.0

### ISSUE THREAD

**Issue by wey-gu** [nebula-contrib/ngbatis#95]
as title

**Comment by CorvusYe** [comment]
I referred to the snapshot workflow and tried to solve it #96

**Comment by CorvusYe** [comment]
Maybe I know why the failure occurs. When the snapshot workflow is completed, v1.1.0 has been deployed. When the release workflow runs, v1.1.0 exists in maven repo, and the release version cannot be rewritten.
It's right ? @Nicole00 

Although the workflow failed, it was published successfully. @wey-gu 
https://search.maven.org/artifact/org.nebula-contrib/ngbatis/1.1.0/jar

https://github.com/nebula-contrib/ngbatis/actions/runs/3570825302/jobs/6002147187
![image](https://user-images.githubusercontent.com/15630211/204738432-68b4034d-db8b-4110-b6d9-016c525e0b54.png)

**Comment by wey-gu** [comment]
Before we perform this locally with ` -e` to get more debug info, I guess it's due to somehow, yesterday, this release was published once for 1.1.0.

```
Error:  Nexus Staging Rules Failure Report
Error:  ==================================
Error:  
Error:  Repository "orgnebula-contrib-1014" failures
Error:    Rule "RepositoryWritePolicy" failures
Error:      * Artifact updating: Repository ='releases:Releases' does not allow updating artifact='/org/nebula-contrib/ngbatis/1.1.0/ngbatis-1.1.0-javadoc.jar'
Error:      * Artifact updating: Repository ='releases:Releases' does not allow updating artifact='/org/nebula-contrib/ngbatis/1.1.0/ngbatis-1.1.0.pom'
Error:      * Artifact updating: Repository ='releases:Releases' does not allow updating artifact='/org/nebula-contrib/ngbatis/1.1.0/ngbatis-1.1.0-tests.jar'
Error:      * Artifact updating: Repository ='releases:Releases' does not allow updating artifact='/org/nebula-contrib/ngbatis/1.1.0/ngbatis-1.1.0-sources.jar'
Error:      * Artifact updating: Repository ='releases:Releases' does not allow updating artifact='/org/nebula-contrib/ngbatis/1.1.0/ngbatis-1.1.0.jar'
Error:  
Error:  
```

![Screenshot 2022-11-30 at 16 09 15](https://user-images.githubusercontent.com/1651790/204741786-615380f7-00dc-4353-acdc-b63145b1b0b7.png)

@CorvusYe could you please try this, to do a release with vername: 1.1.1 or 1.1.0-ga ?

**Comment by CorvusYe** [comment]
@wey-gu 
I created another PR of 1.1.1

**Comment by CorvusYe** [comment]
![image](https://user-images.githubusercontent.com/15630211/204753656-846723ff-a07d-4ead-bf94-6b5451407b1d.png)
Published successfully by snapshot, and release workflow failed again.😂😂

**Comment by wey-gu** [comment]
Now I know why it's failed, it's due to the snapshot was triggered, too, and it's already published to maven.

see https://github.com/nebula-contrib/ngbatis/actions/runs/3581743231/jobs/6025157176

**Comment by wey-gu** [comment]
Yes, we should ensure release not trigger snapshot, maybe we only enable snapshot every night?

**Comment by CorvusYe** [comment]
If different workflows use different maven profiles to publish different versions (snapshots and releases) with the same code, it may be a better practice.

**Comment by wey-gu** [comment]
> If different workflows use different maven profiles to publish different versions (snapshots and releases) with the same code, it may be a better practice.

agreed
